src/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

```diff
@@ -66,23 +58,35 @@ public FileAttributes GetAttributes(ReadOnlySpan<char> path, ReadOnlySpan<char>
                 writeBit = Interop.Sys.Permissions.S_IWOTH;
             }
 
-            if ((_fileStatus.Mode & (int)readBit) != 0 && // has read permission
-                (_fileStatus.Mode & (int)writeBit) == 0)  // but not write permission
-            {
-                attrs |= FileAttributes.ReadOnly;
-            }
+            return ((_fileStatus.Mode & (int)readBit) != 0 && // has read permission
+                (_fileStatus.Mode & (int)writeBit) == 0);     // but not write permission
+        }
+
+        public FileAttributes GetAttributes(ReadOnlySpan<char> path, ReadOnlySpan<char> fileName)
+        {
+            // IMPORTANT: Attribute logic must match the logic in FileSystemEntry
+
+            EnsureStatInitialized(path);
+
+            if (!_exists)
+                return (FileAttributes)(-1);
```

Comment:
Since this has a specific meaning, shouldn't be added to the enum or at least as a constant?

Reply:
Not sure if we can add it to the enum as it is every flag? If we think it won't cause grief I'd rather it be in there. @terrajobst - what do you think on this one? `FileAttributes.Invalid`?